prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -168,12 +183,53 @@ object CustomFHIRFunctions : FhirPathFunctions {
                     changeTimezone(focus, parameters)
                 }
 
+                CustomFHIRFunctionNames.ConvertDateToAge -> {
+                    convertDateToAge(focus)
+                }
+
                 else -> additionalFunctions?.executeFunction(focus, functionName, parameters)
                     ?: throw IllegalStateException("Tried to execute invalid FHIR Path function $functionName")
             }
             )
     }
 
+    /**
+     * Converts the date in the [focus] element to an age in years, months, weeks, or days.
+     * @return an age in years, months, weeks, or days.
+     */
+    fun convertDateToAge(focus: MutableList<Base>): MutableList<Base> {
+        val date = focus.first()
+
+        if (date !is DateType) {
+            throw SchemaException("Must call the convertDateToAge function on a single dateTimeType")
+        }
+
+        val period = Period.between(
+            // have to do plus one for the month because it is expecting 1 based and we get zero based
+            LocalDate.of(date.year, date.month + 1, date.day),
+            LocalDate.now()
+        )
+
+        val age = Age()
+        if (period.years > 1) {
```

Comment:
I'm okay with this for now, but I'm uneasy about us having the logic to decide what the unit should be in this function, my initial thought was that should be passed as a param...